src/library/scala/collection/immutable/TreeMap.scala

```diff
@@ -121,7 +126,7 @@ final class TreeMap[A, +B] private (tree: RB.Tree[A, B])(implicit val ordering:
 
   /** A factory to create empty maps of the same type of keys.
    */
-  override def empty: TreeMap[A, B] = TreeMap.empty[A, B](ordering)
+  override def empty: TreeMap[A, B] = newMap(null)
```

Comment:
same as in `TreeSet`: why create a new instance?

Reply:
Echoing @mkeskells answer: `TreeMap.empty` isn't a singleton, it creates a new `TreeMap` instance with the implicitly supplied `Ordering`.